### PR TITLE
Feature/enhanced help messages

### LINKED ISSUE #237: enhanced instruction context help messages
All instruction messages should contain information about immediate operand length & edge cases

### PR DESCRIPTION
**Pull request by ErikP0**
Adds a range description to every instruction accepting immediate values. closes #237

### CODE REVIEW

**Review by k-lukas — APPROVED** [review]

File: include/arch/riscv/documentation-builder.hpp [hunk 1]
```diff
@@ -125,27 +128,33 @@ class DocumentationBuilder {
...
-   * \param operands A list of all operand names that this instruction takes (in the order of definition)
+   * \param operands A list of all operand names that this instruction takes (in
+   * the order of definition)
    * \return The current DocumentationBuilder for chaining convenience
    */
-  DocumentationBuilder& shortSyntax(
-      StringList operands);
+  DocumentationBuilder& shortSyntax(StringList operands);
+
+  DocumentationBuilder& operandRange(const std::string& name, int range,
```
Comment (goldsborough) [inline include/arch/riscv/documentation-builder.hpp:144]: I don't think this is clang-formatted

File: include/arch/riscv/documentation-builder.hpp [hunk 2]
```diff
@@ -156,14 +165,16 @@ class DocumentationBuilder {
...
    * \return
    */
-  Translateable::TranslateablePtr _optional(const Key& key);
+  TranslateablePtr _optional(const Key& key);
 
-  /** A map to store all components until building to offer arbitrary filling order */
-  std::unordered_map<Key, Translateable::TranslateablePtr, Utility::EnumHash<Key>> _components;
+  /** A map to store all components until building to offer arbitrary filling
+   * order */
+  std::unordered_map<Key, TranslateablePtr, Utility::EnumHash<Key>> _components;
```
Comment (goldsborough) [inline include/arch/riscv/documentation-builder.hpp]: I would use an alias here

File: source/arch/riscv/instruction-context-information.cpp [hunk 3]
```diff
@@ -536,3 +546,13 @@ void InstructionContextInformation::_add(const std::string &key,
                                          const Translateable &value) {
   _table.emplace(key, value);
 }
+
+bool InstructionContextInformation::_isImmediateInstruction(
+    const std::string &mnemonic) const {
+  if (mnemonic.size() >= 2) {
+    auto substring = mnemonic.substr(mnemonic.size() - 2, mnemonic.size());
+    return substring == std::string("iw") || substring.back() == 'i';
```
Comment (goldsborough) [inline source/arch/riscv/instruction-context-information.cpp]: I think you can compare directly with `"iw"`

File: source/arch/riscv/documentation-builder.cpp [hunk 4]
```diff
@@ -100,6 +109,19 @@ DocumentationBuilder &DocumentationBuilder::shortSyntax(StringList operands) {
   return *this;
 }
 
+DocumentationBuilder &DocumentationBuilder::operandRange(
+    const std::string &name, int range, bool isSigned) {
+  // signed range is from -2^(n-1) to 2^(n-1)-1
+  // unsigned range is from 0 to 2^(n)-1
+  int minRange = isSigned ? -(1 << (range-1)) : 0;
```
Comment (goldsborough) [inline source/arch/riscv/documentation-builder.cpp]: I think `lowerBound` and `upperBound` would be clearer here

**Review by goldsborough — APPROVED** [review]

**Pull request merged**